dll/win32/shell32/folders/CRegFolder.cpp

```diff
@@ -772,8 +772,10 @@ HRESULT WINAPI CRegFolder::SetNameOf(HWND hwndOwner, PCUITEMID_CHILD pidl,    /*
 
     if (res == ERROR_SUCCESS)
     {
-        *pPidlOut = ILClone(pidl);
-        return S_OK;
+        hr = S_OK;
+        if (pPidlOut)
+            hr = SHILClone(pidl, pPidlOut);
+        return hr;
```

Comment:
```suggestion
        return (pPidlOut ? SHILClone(pidl, pPidlOut) : S_OK);
```
or, to help with debugging the returned value,
```suggestion
        hr = (pPidlOut ? SHILClone(pidl, pPidlOut) : S_OK);
        return hr;
```